stelaval/assignment1/SinglyLinkedList.java

```diff
@@ -0,0 +1,95 @@
+package assignment1;
+
+public class SinglyLinkedList<T> {
+
+	private Node<T> head;
+
+	public Node<T> getHead() {
+		return head;
+	}
+
+	public boolean isEmpty() {
+		return head == null;
+	}
+
+	/**
+	 * Finds the kth to last element of the singly linked list
+	 */
+	public T getKthToLastElement(int k) {
+		int length = 0;
+		Node<T> currentNode = getHead();
+
+		while (currentNode != null) {
+			currentNode = currentNode.getNext();
+			length++;
+		}
+
+		if (length < k) {
+			try {

```

Comment:
No need to catch it here. Either declare exception in the method (`public T getKthToLastElement(int k) throws Exception`) and catch it in `main`, or use exception that inherits from `RuntimeException` which does not need to be declared.

In this case the best approach would be to throw `IllegalArgumentException` (it is a `RuntimeException`, so no need to declare it in method).